packages/scene-composer/src/components/WebGLCanvasManager.tsx

```diff
@@ -39,7 +39,8 @@ export const WebGLCanvasManager: React.FC = () => {
   const appendSceneNode = useStore(sceneComposerId)((state) => state.appendSceneNode);
   const { gl } = useThree();
   const domRef = useRef<HTMLElement>(gl.domElement.parentElement);
-  const environmentPreset = getSceneProperty(KnownSceneProperty.EnvironmentPreset);
+  const defaultLighting: string | undefined = useViewOptionState(sceneComposerId).defaultLighting;
```

Comment:
Nit: I think it’s useful to have a ‘Maybe’ utility type available in the repo to handle non-existence. Then you can create some super useful type predicates.

```typescript
type Just<T> = T;
type Nothing = null | undefined;
type Maybe<T> = Just<T> | Nothing;

function isJust<T>(maybe: Maybe<T>): maybe is Just<T> {
  return maybe != null;
}

function isNothing<T>(maybe: Maybe<T>): maybe is Nothing {
  return !isJust(maybe);
}
```

This makes null checks a bit cleaner and can improve type safety, depending on the usage.

Reply:
I like the concept. The only pushback that I'd have is that I would like to see us start using `undefined` as it is meant to work in JavaScript - "Hey! Nothing was assigned to this." The problem with grouping it with `null` is that there is the `nullable` concept where we would intentionally assign `null` to a variable in place of a primitive, object, etc... By keeping these seprate, we're able to appropriately determine if a variable initialized without a value (`undefined`) or was intentionally empty (`null`). 

This was left with the `string` | `undefined` in order to improve readability that sometimes (initial scene creation) this value is truly `undefined`. 

Let me know if that doesn't make sense or I misunderstood the feedback. :)

Reply:
Nope, I agree. I don't prefer to use null either. It's simply for the `Nothing` type itself that it includes null, as you can't control things like a library integration.